### PR TITLE
Update Pester to work on PowerShell Core at Windows, Linux, macOS

### PR DESCRIPTION
**Pull request by it-praktyk**
The related issue: 
- #639 
- #838 
- #864
- #870

### CODE REVIEW

**Comment by it-praktyk** [comment]
@dlwyatt, we have added the .travis file. Can it cause that the current CI workflow will not work properly?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+os:
+  - linux
+sudo: required
```
Comment (felixfbecker) [inline .travis.yml]: Is this needed? It's possible to install apt dependencies with `addons.apt`: https://docs.travis-ci.com/user/installing-dependencies/#Adding-APT-Packages
(builds without `sudo` are a lot faster)

Comment (it-praktyk) [reply .travis.yml]: @felixfbecker, thank you for review. Your proposal looks reasonable. Please prepare an update and request a pull request to the branch  [Dev4-1_PSCore_compatibility](https://github.com/pester/Pester/tree/Dev4-1_PSCore_compatibility).

Comment (nohwnd) [reply .travis.yml]: @felixfbecker thanks for that, I did not really know what I am doing when I was putting the travis.yml in place

**Comment by felixfbecker** [comment]
Is there anything left here before this can get merged? I tried it on https://github.com/aaronpowell/ps-nvmw/pull/28 and the tests passed (they don't on master) 🎉

**Comment by nohwnd** [comment]
@it-praktyk the related issue should be closed when this is merged, right? I changed the description to do that. 
Is there anything else that needs to be done before merging this?

**Comment by it-praktyk** [comment]
Please don't close #838 - I'll do it after adding notes in a documentation

**Comment by it-praktyk** [comment]
The pull request #903 is related but an order of merging with the current PR doesn't matter, for the second one rebasing will be needed.

**Comment by it-praktyk** [comment]
@nohwnd, the last build fail was caused by [an issue in Travis CI infrastructure](https://www.traviscistatus.com/incidents/rqf3s76q1rtw).

**Comment by nohwnd** [comment]
@it-praktyk it's in, you did an awesome job. Thanks a lot :)

**Comment by felixfbecker** [comment]
:shipit:

**Comment by felixfbecker** [comment]
Is there anything preventing this from being released?

**Comment by nohwnd** [comment]
@felixfbecker yes we were waiting for #942, but now it seems to be complete, so there should be no more blockers.

**Pull request merged**